lock.go

```diff
@@ -42,13 +46,26 @@ type Locker struct {
 
 // Run runs a callback handler with a Redis lock. It may return ErrLockNotObtained
 // if a lock was not successfully acquired.
-func Run(client RedisClient, key string, opts *Options, handler func() error) error {
+func Run(client RedisClient, key string, opts *Options, handler func()) error {
```

Comment:
why `handler func()` should not return error?

Reply:
Because you need to handle this stuff aside of `Lock/Unlock` errors/flow.

In e.g. when you have error thrown by `Obtain` you will never know in your code it this relates to **lock** or this is your **handler** error.

Reply:
hm.. agree, makes sense